image/image.go

```diff
@@ -172,6 +179,32 @@ func (i *genericImage) getSchema1Manifest() (manifest, error) {
 	return mschema1, nil
 }
 
+// uniqueLayerDigests returns a list of layer digests referenced from a manifest.
+// The list will not contain duplicates; it is not intended to correspond to the "history" or "parent chain" of a Docker image.
+func uniqueLayerDigests(m manifest) []string {
+	var res []string
+	seen := make(map[string]struct{})
+	for _, digest := range m.GetLayers() {
+		if _, ok := seen[digest]; ok {
+			continue
+		}
+		seen[digest] = struct{}{}
+		res = append(res, digest)
+	}
+	return res
+}
+
+// LayerDigests returns a list of layer digests referenced by this image.
+// The list will not contain duplicates; it is not intended to correspond to the "history" or "parent chain" of a Docker image.
+// NOTE: It is essential for signature verification that LayerDigests is computed from the same manifest which is returned by Manifest().
+func (i *genericImage) LayerDigests() ([]string, error) {
+	m, err := i.getSchema1Manifest()
```

Comment:
another question, I'm lost thanks for bearing with me - how do we handle this type of switch https://github.com/projectatomic/skopeo/pull/102/files#diff-0afbed05940006aa6ff131fc915c90b5R16?